plugin/core/views.py

```diff
@@ -1,9 +1,15 @@
-import sublime
+import html
 import linecache
+import mdpopups
+import re
+import sublime
+import textwrap
 from .protocol import Point, Range, Notification, Request
-from .typing import Optional, Dict, Any, Iterable, List
+from .typing import Optional, Dict, Any, Iterable, List, Union
 from .url import filename_to_uri
 
+DOCS_WRAP_WIDTH = 58
```

Comment:
This seems rather aggressive...

![Screenshot 2020-05-12 at 20 23 34](https://user-images.githubusercontent.com/153197/81730958-94be7c80-948e-11ea-8bc3-c155647dee48.png)

Reply:
Should it be configurable?
For example,
have a setting `"documentation_popup_width"` 
and based on the given width, I somehow calculate the `DOCS_WRAP_WIDTH` ?

Reply:
That is the first thing that came to my mind.
Do you have any suggestions.

And yeah, I have forgotten about the JSONDecodeError which closes the AC for some reason...

Reply:
Max width of popup is set to 480px and wrapping makes the content 300px. So there is still some room for having larger wrap threshold.

Of course we can't reliably determine what will fit in 480px but at least we should get closer to it.

Reply:
I don't know how accurate it will be but we can use `view.em_width()` to get character width.

Reply:
```
DOCS_POPUP_WIDTH = 480 # If have a width

# and if I know the width of the a single char, which I know with view.em_width()

# I can calculate the wrap_width.  (but I also should probably account for any margin left | right, but I need to test that) 

# wrap_width = MAX_WIDTH / char_width
#or maybe
# wrap_width = (MAX_WIDTH - margin-left - margin-right) / char_width

def text2html(view: sublime.View, content: str) -> str:
    content = html.escape(content).replace('\n', '<br>').replace('\t', '&nbsp;&nbsp;&nbsp;&nbsp;')
    x_margins = 12
    wrap_width = int((DOCS_POPUP_WIDTH - x_margins) / view.em_width())
    content = "<br>".join(textwrap.wrap(content, width=wrap_width,
                          replace_whitespace=False, break_long_words=False, break_on_hyphens=False))
```

Reply:
Don't try to be pixel perfect as em_width won't give you perfectly accurate width. It's typically a wide of the widest letter so it depends on font style and what not. You probably have to test and eyeball it.

Reply:
I push the commit for calculating the wrap width.
Right now I am using a constant `DOCS_POPUP_WIDTH` which is hard-coded to `480`.

We can add a settings later if we notice that people want to customize the width of the popup